pkg/registry/controller/etcd/etcd_test.go

```diff
@@ -106,7 +113,8 @@ func TestEtcdCreateController(t *testing.T) {
 	if err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
-	key, _ := makeControllerKey(ctx, validController.Name)
+	prefix := path.Join(storage.Helper.PathPrefix, controllerPrefix)
+	key, _ := etcdgeneric.NamespaceKeyFunc(ctx, prefix, validController.Name)
```

Comment:
OK, it's quite unfortunate that our code is structured the way it is, but I think to make the best of it:

add to the testetcd package something like ```func Key(in string) out string```. Purpose of this function is to take a key as you would give it to an etcd helper, and return a key as you would give it to an etcd cluster (or fake etcd). Then this code shouldn't change, and instead you'd change the next line to ```resp, err := fakeClient.Get(testetcd.Key(key), false, false)```.

Also we will make the rule that the cluster-wide prefix is added by EtcdHelper, and the keys that go into etcd helper do not have it.